compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/converters/ZipConverter.java

```diff
@@ -55,9 +55,9 @@ private static Path pathWithinZip(URI pathToZip) {
                     pathToZip.getQuery(), pathToZip.getFragment());
             initFS(pathInZip);
             return Paths.get(pathInZip);
-        } catch (URISyntaxException ignore) {
+        } catch (URISyntaxException ex) {
+            return Paths.get(pathToZip);
```

Comment:
This does not seems correct. The exception is swallowed and we are not failing when an exception is thrown. What is the reason to return from here?

Reply:
Here we are trying to read the balo which is inside the package zip. An exception can occur when creating the URI needed to create the zip file system provider which will be used to read the content inside the zip/jar file.  So if such an error occurs, we just return the path to the zip/jar file.

Reply:
Added a comment with a proper explanation with 6744db0